src/librustc_lint/bad_style.rs

```diff
@@ -288,11 +288,8 @@ impl LateLintPass for NonSnakeCase {
     }
 
     fn check_pat(&mut self, cx: &LateContext, p: &hir::Pat) {
-        if let &PatKind::Binding(_, ref path1, _) = &p.node {
-            // Exclude parameter names from foreign functions (they have no `Def`)
```

Comment:
@eddyb 
>Bloody windows.

These comment and check were here for a reason - to avoid checking parameter names in foreign functions.
WinAPI, for example, uses totally different idiomatic style than Rust and now large part of imports from WinAPI are going show up as errors in buildbot logs.